### DISCUSSION TITLE
Download LOB apps from Store for Business

### DISCUSSION THREAD

**Discussion by myokeeh** [microsoft/winget-cli#4780] [Q&A] [answered]
There's a recent post from Microsoft that implied I can use winget to download apps that were available from Store for Business. The Store for Business site has been taken down so I can't download there.

The apps are LOB apps with offline licenses.

Is there something else I need or is the post incorrect?

Edit:
Adding link to post for reference:
https://techcommunity.microsoft.com/t5/intune-customer-success/downloading-microsoft-store-apps-using-windows-package-manager/ba-p/4208020

**Comment by denelon** [comment, marked as answer]
WinGet has a "[Download](https://learn.microsoft.com/windows/package-manager/winget/download)" command for packages in configured sources including the Microsoft Store.

**Reply by myokeeh** [reply]
@denelon Yes. That's what I keep trying but I get an error.

```
winget download --id [Store ID Redacted]
Failed when searching source: msstore
An unexpected error occurred while executing the command:
0x8a150044 : The rest API endpoint is not found.
```

Reading through that link you provided, I see `--authentication-account`. I put in my work email and still results in the same error above. I also tried putting `--authentication-mode interactive` and still get the same error.

**Reply by denelon** [reply]
I don't think I've seen that before... I hope it's just a transient network error.

Do you get different results for `winget search <package Id> -s msstore`?

**Reply by myokeeh** [reply]
To clarify, when I said put in my work email, it doesn't prompt me to login. No login window pops up. 

Running the command above results in this erro:

```
An unexpected error occurred while executing the command:
0x8a150044 : The rest API endpoint is not found.
```

To an extent, I expect this because the app is what is probably considered "private" since it's only available to our tenant. 

The problem I'm bringing up to the Intune team in that post I referenced above is it seems like there's something missing about what they're saying. They took out the ability to download app packages from the Store for Business website (it's now down), but they claim I should still be able to get packages using winget.

**Reply by denelon** [reply]
The login is only associated with the requirement for offline distribution of a store package requiring a license file for offline installation. Downloads are coming from the Microsoft Store as opposed to the "Store for Business".

**Reply by myokeeh** [reply]
@denelon  Yeah. So I don't know what the Intune group is missing in the grand scheme of things. They're certainly pointing their finger over here. Does the "Store for Business" need to be a specified as a source somehow? If it does, did they forget to explain how to do that?

**Reply by denelon** [reply]
That's not a thing as far as I'm aware. I don't have all the information about all the transition plans. On the WinGet side of the world, we were focusing on a mechanism to support offline distribution of store apps, and we still have work to deal with licenses that could be added to a WinGet package manifest in a private WinGet REST source. I think there are potentially a few other gaps, but I'm not aware of any way to get packages from a "store for business" installed or downloaded using WinGet. Is there a public URL or something you could point me to?

**Reply by myokeeh** [reply]
It sounds like you and I are on the same page. I was aware of the work your group has been working on. I don't know of a way to define the "store for business" source or URL--that has never been something I needed. I was assuming that maybe the author(s) of that post may have an incomplete picture or misunderstanding of what's available via winget.

**Reply by denelon** [reply]
I re-read the article, and I think it's just not clear that there is no notion of downloading LOB apps uploaded to the store for business. Only apps in the Microsoft Store are available. If there are LOB apps, they would need to be made available via a private source for WinGet to be able to act on them. We're still closing gaps on Entra ID authentication for private REST sources, and we still have work to do for downloading apps requiring Entra ID from enterprise managed storage. Initially, we're looking at (ACR) Azure Container Registry as a source an enterprise could use to host the "installers" for distribution via WinGet.
